src/Perlang.Tests/Interpreter/Resolution/ResolverTest.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Perlang.Interpreter;
+using Perlang.Interpreter.Resolution;
+using Perlang.Parser;
+using Xunit;
+
+namespace Perlang.Tests.Interpreter.Resolution
+{
+    public class ResolverTest
+    {
+        [Fact(Skip = "Known to be broken, will be fixed shortly on another branch")]
+        public void VisitVarStmt_defines_variable_with_correct_TypeReference()
+        {
+            // Act
+            (Stmt singleStatement, Resolver resolver) = ScanParseAndResolveSingleStatement(@"
+                var i: int = 123456;
+            ");
+
+            // Assert. We want to ensure that the proper TypeReference is being used in the VariableBindingFactory
+            // which has been created at this point, since using the TypeReference from the initializer instead of the
+            // variable will work incorrectly with e.g. integer expansion (`var l: long = 123` - it's critical that
+            // TypeReference becomes `long` here and not `int` or `short`)
+            Assert.IsType<Stmt.Var>(singleStatement);
+            Assert.True(resolver.Globals.ContainsKey("i"));
+            Assert.Equal(((Stmt.Var)singleStatement).TypeReference, ((VariableBindingFactory)resolver.Globals["i"]).TypeReference);
+        }
```

Comment:
(This tests will be fixed to no longer fail in #70, but it made sense to include it here already since it serves as an example of how the new `ScanAndParse` code is used from tests.)